tests/unit_tests/dft/include/compute_inplace.hpp

```diff
@@ -0,0 +1,157 @@
+/***************************************************************************
+*  Copyright (C) Codeplay Software Limited
+*  Licensed under the Apache License, Version 2.0 (the "License");
+*  you may not use this file except in compliance with the License.
+*  You may obtain a copy of the License at
+*
+*      http://www.apache.org/licenses/LICENSE-2.0
+*
+*  For your convenience, a copy of the License has been included in this
+*  repository.
+*
+*  Unless required by applicable law or agreed to in writing, software
+*  distributed under the License is distributed on an "AS IS" BASIS,
+*  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*  See the License for the specific language governing permissions and
+*  limitations under the License.
+*
+**************************************************************************/
+
+#ifndef ONEMKL_COMPUTE_INPLACE_HPP
+#define ONEMKL_COMPUTE_INPLACE_HPP
+
+template <dft::precision precision, dft::domain domain>
+int DFT_Test<precision, domain>::test_in_place_buffer() {
+    if (!init()) {
+        return test_skipped;
+    }
+
+    descriptor_t descriptor{ size };
+    descriptor.set_value(oneapi::mkl::dft::config_param::PLACEMENT,
+                         oneapi::mkl::dft::config_value::INPLACE);
+
+    const size_t container_size = domain == dft::domain::REAL ? conjugate_even_size : size;
+
+    sycl::buffer<InputType, 1> inout_dev{ sycl::range<1>(container_size) };
+    std::vector<InputType> out_host(container_size);
+
+    copy_to_device(sycl_queue, input, inout_dev);
+
+    commit_descriptor(descriptor, sycl_queue);
+
+    try {
+        dft::compute_forward<std::remove_reference_t<decltype(descriptor)>, InputType>(descriptor,
```

Comment:
```suggestion
        dft::compute_forward<descriptor_t>, InputType>(descriptor,
```
and similar elsewhere.